packages/agent/src/hooks/useAcceptDidCommPresentation.ts

```diff
@@ -157,9 +157,33 @@ export function useAcceptDidCommPresentation(proofExchangeId: string) {
     },
   })
 
+  const { mutateAsync: declineMutateAsync } = useMutation({
+    mutationKey: ['declineDidCommPresentation', proofExchangeId],
+    mutationFn: async () => {
+      const presentationDeclined$ = agent.events
+        .observable<ProofStateChangedEvent>(ProofEventTypes.ProofStateChanged)
+        .pipe(
+          // Correct record with id and state
+          filter(
+            (event) =>
+              event.payload.proofRecord.id === proofExchangeId &&
+              [ProofState.Declined].includes(event.payload.proofRecord.state)
+          ),
+          // 10 seconds to complete exchange
+          timeout(10000),
+          first()
```

Comment:
I think the method call will wait with returning until this state is reached, so i think we can remove it

Reply:
But it's quite nice to have this functionality right just like the accept action. And the state is updated locally in afj right for the record or does it have to wait for the external party?